pkg/client/client_test.go

```diff
@@ -59,133 +66,248 @@ func TestApplySingle(t *testing.T) {
+	t.Run("updates of Services", func(t *testing.T) {
+
 		// given
 		namespacedName := types.NamespacedName{Namespace: "toolchain-host-operator", Name: "registration-service"}
 
-		t.Run("when using forceUpdate=true, it should not update when specs are same", func(t *testing.T) {
-			// given
-			cl, _ := newClient(t, s)
-			obj := defaultService.DeepCopy()
-			_, err := cl.CreateOrUpdateObject(obj, true, nil)
-			require.NoError(t, err)
-			originalGeneration := obj.GetGeneration()
-
-			// when updating with the same obj again
-			createdOrChanged, err := cl.CreateOrUpdateObject(obj, true, nil)
-
-			// then
-			require.NoError(t, err)
-			assert.False(t, createdOrChanged) // resource was not updated on the server, so returned value is `false`
-			updateGeneration := obj.GetGeneration()
-			assert.Equal(t, originalGeneration, updateGeneration)
+		t.Run("as corev1 objects", func(t *testing.T) {
+
+			t.Run("when using forceUpdate=true", func(t *testing.T) {
+
+				t.Run("it should not update when specs are same", func(t *testing.T) {
+					// given
+					cl, _ := newClient(t, s)
+					obj := defaultService.DeepCopy()
+					_, err := cl.CreateOrUpdateObject(obj, true, nil)
+					require.NoError(t, err)
+					originalGeneration := obj.GetGeneration()
+
+					// when updating with the same obj again
+					createdOrChanged, err := cl.CreateOrUpdateObject(obj, true, nil)
+
+					// then
+					require.NoError(t, err)
+					assert.False(t, createdOrChanged) // resource was not updated on the server, so returned value is `false`
+					updateGeneration := obj.GetGeneration()
+					assert.Equal(t, originalGeneration, updateGeneration)
+				})
+
+				t.Run("it should not update when specs are same except ClusterIP", func(t *testing.T) {
+					// given
+					cl, _ := newClient(t, s)
+					obj := defaultService.DeepCopy()
+					_, err := cl.CreateOrUpdateObject(obj, true, nil)
+					require.NoError(t, err)
+					originalGeneration := obj.GetGeneration()
+					obj.Spec.ClusterIP = "" // modify for version to update
+					// when updating with the same obj again
+					createdOrChanged, err := cl.CreateOrUpdateObject(obj, true, nil)
+
+					// then
+					require.NoError(t, err)
+					assert.False(t, createdOrChanged) // resource was not updated on the server, so returned value is `false`
+					updateGeneration := obj.GetGeneration()
+					assert.Equal(t, originalGeneration, updateGeneration)
+					assert.Equal(t, defaultService.Spec.ClusterIP, obj.Spec.ClusterIP)
+				})
+
+				t.Run("it should update when specs are different", func(t *testing.T) {
+					// given
+					cl, _ := newClient(t, s)
+					obj := defaultService.DeepCopy()
+					_, err := cl.CreateOrUpdateObject(obj, true, nil)
+					require.NoError(t, err)
+					originalGeneration := obj.GetGeneration()
+
+					// when updating with the modified obj
+					modifiedObj := modifiedService.DeepCopy()
+					modifiedObj.Spec.ClusterIP = ""
+					modifiedObj.ObjectMeta.Generation = obj.GetGeneration()
```

Comment:
why do you copy the Generation? this shouldn't be needed
```suggestion
```

Reply:
Well, that was part of the existing test ;) There was simply a change on these lines because I moved the test into a subtest. 
But you're right, this line is not needed and I'll remove it ;)

Reply:
done in 83e733b